store/store.go

```diff
@@ -892,7 +921,11 @@ func (s *Store) Download(name string, downloadInfo *snap.DownloadInfo, pbar prog
 var download = func(name, url string, user *auth.UserState, s *Store, w io.Writer, pbar progress.Meter) error {
 	client := &http.Client{}
 
-	resp, err := s.doRequest(client, "GET", url, halJsonContentType, "", nil, user)
+	reqOptions := &requestOptions{
+		Method: "GET",
+		Url:    url,
```

Comment:
I can restore it, but server is not supposed to require it (there is no hal+json response expected to be returned in a download). I guess this inherited the accept at the time because of the setStoreHeaders helper set it for all.